game-core/src/main/java/games/strategy/triplea/ui/EconomyPanel.java

```diff
@@ -76,22 +93,40 @@ private synchronized void loadData() {
       gameData.acquireReadLock();
       try {
         final List<PlayerID> players = getPlayers();
-        final Collection<String> alliances = getAlliances();
-        collectedData = new String[players.size() + alliances.size()][statsResource.length + 1];
+        final Map<String, Set<PlayerID>> allianceMap = getAllianceMap();
+        collectedData = new String[players.size() + allianceMap.size()][resourceStats.size() + 1];
         int row = 0;
+        final Map<PlayerID, IntegerMap<Resource>> resourceIncomeMap = new HashMap<>();
         for (final PlayerID player : players) {
           collectedData[row][0] = player.getName();
-          for (int i = 0; i < statsResource.length; i++) {
-            collectedData[row][i + 1] =
-                statsResource[i].getFormatter().format(statsResource[i].getValue(player, gameData));
+          final IntegerMap<Resource> resourceIncomes = AbstractEndTurnDelegate.findEstimatedIncome(player, gameData);
+          resourceIncomeMap.put(player, resourceIncomes);
+          for (int i = 0; i < resourceStats.size(); i++) {
+            final ResourceStat resourceStat = resourceStats.get(i);
+            final Resource resource = resourceStat.resource;
+            final double quantity = resourceStat.getValue(player, gameData);
+            final StringBuilder text = new StringBuilder(resourceStat.getFormatter().format(quantity) + " (");
+            if (resourceIncomes.getInt(resource) >= 0) {
+              text.append("+");
+            }
+            text.append(resourceIncomes.getInt(resource) + ")");
+            collectedData[row][i + 1] = text.toString();
           }
           row++;
         }
-        for (final String alliance : alliances) {
-          collectedData[row][0] = alliance;
-          for (int i = 0; i < statsResource.length; i++) {
-            collectedData[row][i + 1] =
-                statsResource[i].getFormatter().format(statsResource[i].getValue(alliance, gameData));
+        for (final Entry<String, Set<PlayerID>> alliance : allianceMap.entrySet()) {
+          collectedData[row][0] = alliance.getKey();
+          for (int i = 0; i < resourceStats.size(); i++) {
+            final ResourceStat resourceStat = resourceStats.get(i);
+            final double quantity = resourceStat.getValue(alliance.getKey(), gameData);
+            final StringBuilder text = new StringBuilder(resourceStat.getFormatter().format(quantity) + " (");
+            final int income = alliance.getValue().stream()
+                .mapToInt(p -> resourceIncomeMap.get(p).getInt(resourceStat.resource)).sum();
+            if (income >= 0) {
```

Comment:
deja-vu

Please do extract to a function,most of the bodies of these for loops look about the same, can be pretty cleanly parameterized with resourceStat and the conditional check value (eg: `resourceIncomes.getInt(resource) >= 0`)

Reply:
Extracted helper method.